### ISSUE TITLE
Enable SQL features selectively

### ISSUE THREAD

**Issue by vszakd** [sqlparser-rs/sqlparser-rs#1201]
Hello,
given the breadth of the parser, it would be nice if it would be possible to selectively enable features on an opt-in basis.

Let's imagine an application where only a subset of SQL is supported. In this case, the application would need to check for every possible unsupported feature manually (e.g. if I don't support windowing in queries, I would need to check if the `named_window` field in the `Select` gets populated and abort).

It would be interesting, instead, to decide beforehand which feature set is to be supported, so that only "valid" SQL gets parsed and arrives to the application.

Thanks.

**Comment by alamb** [comment]
I agree this is interesting, though I am not sure how easy it would be to add to the current crate
